src/lib/components/ExplorerLink.tsx

```diff
@@ -42,33 +43,48 @@ type CommonExplorerLinkProps = FlexProps & {
   isReadOnly?: boolean;
   leftIcon?: ReactNode;
   openNewTab?: boolean;
+  queryParams?: Record<string, string>;
   rightIcon?: ReactNode;
   showCopyOnHover?: boolean;
   textFormat?: TextFormat;
+  textLabel?: string;
   textVariant?: TextProps["variant"];
-  type: LinkType;
   value: string;
 };
 
-type ContractAddressExplorerLinkProps = CommonExplorerLinkProps & {
+type ContractAddressExplorerLinkProps = Omit<
+  CommonExplorerLinkProps,
+  "textLabel" | "type"
+> & {
   textLabel: Option<string>;
   type: "evm_contract_address";
 };
 
+type FunctionNameExplorerLinkProps = Omit<
+  CommonExplorerLinkProps,
+  "queryParams" | "type" | "value"
+> & {
+  queryParams: Record<string, string>;
+  type: "function_name";
+  value: Option<string>;
+};
+
```

Comment:
it's no need to omit, you can just a intersection like
```
type FunctionNameExplorerLinkProps = CommonExplorerLinkProps & {
  queryParams: Record<string, string>;
  type: "function_name";
  value: Option<string>;
};
```
It's because the intersection automatically handles property overriding.